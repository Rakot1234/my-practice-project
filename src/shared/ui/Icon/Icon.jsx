import React, { PureComponent } from 'react';
import './Icon.scss';
import PropTypes from 'prop-types';
import cx from 'classnames';

class Icon extends PureComponent {
    static propTypes = {
        icon: PropTypes.string.isRequired,
        className: PropTypes.string,
        onClick: PropTypes.func
    };

    static defaultProps = {
        className: ''
    }

    render() {
        const { icon, className, onClick } = this.props;

        return (
            <div className={cx('icon', className)}>
                <img
                    src={icon}
                    alt=""
                    className={cx('icon__image')}
                    onClick={onClick}
                />
            </div>
        );
    }
};

export default Icon;
