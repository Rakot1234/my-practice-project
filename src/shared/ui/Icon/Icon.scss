@import '../../../styles/variables.scss';

.icon {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    &__image {
        max-width: 100%;
        max-height: 100%;
    }
};
