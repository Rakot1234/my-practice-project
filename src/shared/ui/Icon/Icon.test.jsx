import React from 'react';
import { shallow } from 'enzyme';
import Icon from './Icon';

describe('<Icon />', () => {
    it('Renders icon component', () => {
        const wrapper = shallow(<Icon icon="some-icon.png" />);
        expect(wrapper).toMatchSnapshot();
    });
});