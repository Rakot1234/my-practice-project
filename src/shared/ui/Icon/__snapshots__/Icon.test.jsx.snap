// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Renders icon component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    className=""
    icon="some-icon.png"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt=""
        className="icon__image"
        src="some-icon.png"
      />,
      "className": "icon",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "",
        "className": "icon__image",
        "onClick": undefined,
        "src": "some-icon.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt=""
          className="icon__image"
          src="some-icon.png"
        />,
        "className": "icon",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "icon__image",
          "onClick": undefined,
          "src": "some-icon.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
